core/src/main/java/com/opensymphony/xwork2/conversion/impl/StringConverter.java

```diff
@@ -65,8 +65,12 @@ protected String convertToString(Locale locale, Object value) {
         if (Number.class.isInstance(value)) {
             NumberFormat format = NumberFormat.getNumberInstance(locale);
             format.setGroupingUsed(false);
-            if (Double.class.isInstance(value) || BigDecimal.class.isInstance(value)) {
-                format.setMinimumFractionDigits(1);
+            if (BigDecimal.class.isInstance(value) || Double.class.isInstance(value) || Float.class.isInstance(value)) {
+                format.setMaximumFractionDigits(Integer.MAX_VALUE);
+                // TODO: delete this if statement when jdk fixed java.text.NumberFormat.format's behavior with Float
+                if (Float.class.isInstance(value)) {
+                    value = Double.valueOf(value.toString());
```

Comment:
I have just one doubt: re-assigning to an input parameter

Reply:
:) you're right! I added a test in a commit which confirms that StringConverter does not change input parameter `value`.

Reply:
I meant that this isn't a problem from compiler point-of-view as parameters are always local (compiler makes a copy of them), it's rather a bad style :)

Reply:
Oh :) I thought you are in doubt because we pass the value parameter by type `Object`.

Thanks for learning this to me! I researched and it seems there are two best practices ; using a defined local variable, or returning from method at that line. what do you recommend?

Reply:
I would use a dedicated local variable in this case.

Reply:
Thanks! I think I fixed it in my another commit ; I'm sorry for these a lot of commits :/